meson build system on 3.9.0 fails under NixOS

## Bug Report

### Your environment

- Operating system:  NixOS 21.05
- Node version: 14.17.6
- npm version: 6.14.15
- gcc/clang version:  10.3.0
- mediasoup version: 3.9.0
- mediasoup-client version: N/A


### Issue description

```
jia:/tmp$ git clone https://github.com/versatica/mediasoup.git
Cloning into 'mediasoup'...
remote: Enumerating objects: 55093, done.
remote: Counting objects: 100% (3907/3907), done.
remote: Compressing objects: 100% (2212/2212), done.
remote: Total 55093 (delta 1966), reused 3151 (delta 1582), pack-reused 51186
Receiving objects: 100% (55093/55093), 161.81 MiB | 7.98 MiB/s, done.
Resolving deltas: 100% (34953/34953), done.

jia:/tmp$ nix-shell --pure -p nodejs python3 python3Packages.pip wget

[nix-shell:/tmp]$ cd mediasoup/

[nix-shell:/tmp/mediasoup]$ npm install

> mediasoup@3.9.0 postinstall /tmp/mediasoup
> node npm-scripts.js postinstall

npm-scripts.js [INFO] running task "postinstall"
npm-scripts.js [INFO] executing command: node npm-scripts.js worker:build
npm-scripts.js [INFO] running task "worker:build"
npm-scripts.js [INFO] executing command: make -C worker
make: Entering directory '/tmp/mediasoup/worker'
# Updated pip and setuptools are needed for meson
# `--system` is not present everywhere and is only needed as workaround for Debian-specific issue (copied from
# https://github.com/gluster/gstatus/pull/33), fallback to command without `--system` if the first one fails.
/nix/store/32q6ryrb860sksdi1al5djg3pgcpq92l-python3-3.8.11/bin/python3 -m pip install --system --target=/tmp/mediasoup/worker/out/pip pip setuptools || \
        /nix/store/32q6ryrb860sksdi1al5djg3pgcpq92l-python3-3.8.11/bin/python3 -m pip install --target=/tmp/mediasoup/worker/out/pip pip setuptools || \
        echo "Installation failed, likely because PIP is unavailable, if you are on Debian/Ubuntu or derivative please install the python3-pip package"
/nix/store/32q6ryrb860sksdi1al5djg3pgcpq92l-python3-3.8.11/bin/python3: No module named pip
/nix/store/32q6ryrb860sksdi1al5djg3pgcpq92l-python3-3.8.11/bin/python3: No module named pip
Installation failed, likely because PIP is unavailable, if you are on Debian/Ubuntu or derivative please install the python3-pip package
# Install `meson` and `ninja` using `pip` into custom location, so we don't depend on system-wide installation.
/nix/store/32q6ryrb860sksdi1al5djg3pgcpq92l-python3-3.8.11/bin/python3 -m pip install --upgrade --target=/tmp/mediasoup/worker/out/pip meson ninja
/nix/store/32q6ryrb860sksdi1al5djg3pgcpq92l-python3-3.8.11/bin/python3: No module named pip
make: *** [Makefile:43: meson-ninja] Error 1
make: Leaving directory '/tmp/mediasoup/worker'
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mediasoup@3.9.0 postinstall: `node npm-scripts.js postinstall`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mediasoup@3.9.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jk/.npm/_logs/2021-10-13T10_52_57_200Z-debug.log
```

This happens because PYTHONPATH is being completely replaced in `worker/Makefile`, and NixOS uses it;
replacing line 28 with:
```
PYTHONPATH1 := ${PYTHONPATH}
export PYTHONPATH = $(PIP_DIR):${PYTHONPATH1}
```
fixes that.  Other problems exist which I'm looking at now.

--- a/worker/Makefile
+++ b/worker/Makefile
@@ -25,7 +25,7 @@ else
     export NINJA = $(PIP_DIR)/bin/ninja
 endif
 # Instruct Python where to look for modules it needs, such that `meson` actually runs from installed location.
-export PYTHONPATH = $(PIP_DIR)
+export PYTHONPATH := $(PIP_DIR):${PYTHONPATH}
 
 .PHONY:	\
 	default meson-ninja setup clean clean-pip clean-subprojects clean-all mediasoup-worker xcode lint format test tidy \
@@ -42,7 +42,7 @@ ifeq ($(wildcard $(PIP_DIR)),)
 		$(PYTHON) -m pip install --target=$(PIP_DIR) pip setuptools || \
 		echo "Installation failed, likely because PIP is unavailable, if you are on Debian/Ubuntu or derivative please install the python3-pip package"
 	# Install `meson` and `ninja` using `pip` into custom location, so we don't depend on system-wide installation.
-	$(PYTHON) -m pip install --upgrade --target=$(PIP_DIR) meson ninja
+	$(PYTHON) -m pip install --upgrade --target=$(PIP_DIR) --no-binary ":all:" meson ninja
 endif
 
 setup: meson-ninja

--- a/worker/Makefile
+++ b/worker/Makefile
@@ -14,6 +14,8 @@ DOCKER ?= docker
 PIP_DIR = $(MEDIASOUP_OUT_DIR)/pip
 MESON ?= $(PIP_DIR)/bin/meson
 MESON_ARGS ?= ""
+# Workaround for NixOS and Guix that don't work with pre-built binaries, see:
+# https://github.com/NixOS/nixpkgs/issues/142383.
 PIP_BUILD_BINARIES = $(shell [ -f /etc/NIXOS -o -d /etc/guix ] && echo "--no-binary :all:")
 
 # Disable `*.pyc` files creation.

--- a/worker/Makefile
+++ b/worker/Makefile
@@ -14,6 +14,7 @@ DOCKER ?= docker
 PIP_DIR = $(MEDIASOUP_OUT_DIR)/pip
 MESON ?= $(PIP_DIR)/bin/meson
 MESON_ARGS ?= ""
+PIP_BUILD_BINARIES = $(shell [ -f /etc/NIXOS -o -d /etc/guix ] && echo "--no-binary :all:")
 
 # Disable `*.pyc` files creation.
 export PYTHONDONTWRITEBYTECODE = 1
@@ -42,7 +43,7 @@ ifeq ($(wildcard $(PIP_DIR)),)
 		$(PYTHON) -m pip install --target=$(PIP_DIR) pip setuptools || \
 		echo "Installation failed, likely because PIP is unavailable, if you are on Debian/Ubuntu or derivative please install the python3-pip package"
 	# Install `meson` and `ninja` using `pip` into custom location, so we don't depend on system-wide installation.
-	$(PYTHON) -m pip install --upgrade --target=$(PIP_DIR) --no-binary ":all:" meson ninja
+	$(PYTHON) -m pip install --upgrade --target=$(PIP_DIR) $(PIP_BUILD_BINARIES) meson ninja
 endif
 
 setup: meson-ninja

